### DISCUSSION TITLE
EnableSolarPasstrough

### DISCUSSION THREAD

**Discussion by Adminius** [hoylabs/OpenDTU-OnBattery#103] [General]
Hi, can you please explain more precisely what EnableSolarPasstrough setting does?
a) PV Power 300W, battery not full, household consumption 100W, Inverter & Battery Power?
b) PV Power 300W, battery not full, household consumption 400W, Inverter & Battery Power?

c) PV Power 300W, battery full, household consumption 100W, Inverter & Battery Power?
d) PV Power 300W, battery full, household consumption 400W, Inverter & Battery Power?
e) PV Power 300W, battery empty, household consumption 400W, Inverter & Battery Power?

f) PV Power 50W, battery not full, household consumption 400W, Inverter & Battery Power?
g) PV Power 50W, battery full, household consumption 400W, Inverter & Battery Power?
h) PV Power 50W, battery empty, household consumption 400W, Inverter & Battery Power?

**Comment by berni2288** [comment]
Hi @Adminius 
just saw your message.

I've implemented this feature because of the following thoughts:

- When you store and drain energy from the battery, this round trip will cost you 5-10% of the energy because the efficieny is not 100%
- The lifespan of a battery is not infinite. Every charging cycle (0 - 100%) decreases the battery capacity, even tho a LifePo4 battery can be used up to 6000 charging cycles.
- Each sun watt hour is very valueable for me and I want to lose as less energy as possible.

I'm not a battery expert, but based on the given facts, I thought that it's smarter to let the Hoymiles inverter directly use all the energy, the Victron MPPT charger is currently sending out from the solar panels, instead of charging the battery first with it and then using the energy later when the start threshold is reached.

- Voltage or SOC threshold is below start threshold (= battery not full yet): The limit will not be set higher than the  current Victron Watt output.
- Voltage is below all thresholds (= battery almost empty): All energy will be stored in the battery. Hoymiles limit = 0 Watt; Stopped
- Voltage or SOC Threshold reaches the start threshold (= battery full): max Hoymiles limit = Powermeter Watts (current household usage) until stop threshold is reached again.

**Comment by Adminius** [comment]
> Voltage or SOC Threshold reaches the start threshold (= battery full): max Hoymiles limit = Powermeter Watts (current household usage) until stop threshold is reached again.

This is clear

> Voltage is below all thresholds (= battery almost empty): All energy will be stored in the battery. Hoymiles limit = 0 Watt; Stopped

Ok, but also "energy wasting". I would set "Empty limit" a little bit higher, let say 3.00V/cell (~10% SoC) and use available power for Household. Only if there more power from victron available than household needs, Battery will be chargerd. -> Less cycles, higher system efficiency, but more often empty relatively emptry battery (that is why higher empty limit" not stay at 0% SoC)

> Voltage or SOC threshold is below start threshold (= battery not full yet): The limit will not be set higher than the current Victron Watt output.

Not higher than Victron produces, but also not higher than household consuption, right?
- Victron 100W, Household 300W => Hoymiles 100W, 0W Battery
- Victron 300W, Household 200W => Hoymiles 200W, 100W charging Battery until "full threshold", right?
- Victron 300W, Household 200W, Battery "full" => Hoymiles 200W, 100W wasting or Hoymiles 300W (=100W free feeding)
free feeding is ok for me, this energy can be used by neigbors, but it is not zero feed any more (maybe do it as an option?)

**Reply by berni2288** [reply]
Yes all the things are right and then Powerlimiter currently never sets the limit higher than the total household power consumption (reported by the Power meter).

I've already thought about making disabling the zero energy export to the public grid when the battery is full. But this might be a bit tricky tho since the Victron MPPT will maybe stop charging once the battery voltage is too high.
But maybe a new threshold setting will. For example that you can define a SOC of 95% or a voltage of 53V, and above that threshold all power from the Victron would go into the grid (same as current Direct Solar passtrough feature, but ignoring the actual household consumption).

**Reply by helgeerbe** [reply]
Victron MPPT stops charging if battery is full (state floating) and is reducing current and voltage if it is nearly full (state absorption).
Not to waste any energy you have to operate while MPPT is in bulk state.

**Comment by helgeerbe** [comment]
Hi @berni2288,

after 2 sunny days I'm real happy with your SolarPasstrhrough idea. I can use indeed much more energy, if it is consumed directly. Big Thank you from my side.

**Comment by Technikfreak72** [comment]
But why not feeding the household even the battery is not fully loaded?
Example: Morning hours 150W solar input, household 300W. SOC 30%. I trink it would be better to consume 150W from battery.

**Reply by schlimmchen** [reply]
@Technikfreak72 Have you checked the actual behavior of the current release? As far as I can tell this is indeed implemented as you described. You have to select "Empty at night" discharge strategy, and in your example, stop threshold must be lower than 30%.

**Reply by Technikfreak72** [reply]
I trink always when the household needs more power then it is provided by the solar input the battery has to deliver it, all other options only make sense with very small batteries

**Reply by schlimmchen** [reply]
You can always set your start limit only slightly above your stop limit. In that case the battery will practically always be used. I dont know what happens if you set the limits to the same value or the stop limit larger than the start value?!
